examples/HILO_Symbol_Plotting.py

```diff
@@ -0,0 +1,158 @@
+"""
+=============================
+MSLP and 1000-500 hPa Thickness  with High and Low Symbols
+=============================
+
+Plot MSLP, calculate and plot 1000-500 hPa thickness, and plot H and L markers.
+
+Beyond just plotting a few variables, in the example we use functionality
+from the scipy module to find local maximum and minimimum values within the
+MSLP field in order to plot symbols at those locations.
+"""
+
+###############################
+# Imports
+import numpy as np
+import matplotlib.pyplot as plt
+from metpy.units import units
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+from datetime import datetime
+from netCDF4 import num2date
+from siphon.ncss import NCSS
+from scipy.ndimage import gaussian_filter
+
+###############################
+# Function for finding and plotting max/min points
+
+
+def plot_HILO_points(lon, lat, data, hilo, nsize, transform=None):
+    '''
+    This function will find and plot relative maximum and minimum for a 2D grid. The function
+    is designed to plot an 'H' for maximum values (e.g., High pressure) and an 'L' for minimum
+    values (e.g., low pressue). It first uses a one standard deviation gaussian filter to find
+    a synoptic scale HILO value, then uses the scipy.ndimage filters for maximum or minimum
+    values, finally the HILO symbol is plotted on the current axes with the smoothed integer
+    value of the HILO.
+
+    lon = plotting longitude values (2D)
+    lat = plotting latitude values (2D)
+    data = 2D data that you wish to plot the HILO placement
+    hilo = Either a value of 'hi' for Maximum Values or 'lo' for Minimum Values
```

Comment:
Should we fully spell this out? `type='high', type='low'`?

Reply:
Okay, so I restructured the function to plot arbitrary mix/min points. This would be a better starting point to a function for MetPy.